graph/patterns/namespace.md

```diff
@@ -0,0 +1,125 @@
+# Namespace
+
+Microsoft Graph API Design Pattern
+
+### *The Namespace provides the ability to organize resource definitions together into a logical set.*
+
+## Problem
+
+When building a complex offering API designers may need to model many different
+resources and their relationships. For better user experience and
+discoverability related API elements need to be grouped together.  
+
+
+## Solution
+
+API designers can use the Namespace attribute of the CSDL schema to declare a
+namespace and logically organize related API entities in the Graph metadata.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ XML
+<Schema Namespace="microsoft.graph.{namespace}">
+...
+</Schema>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A public namespace must have "microsoft.graph.” prefix and be presented in camel
+case, i.e microsoft.graph.myNamespace.
+
+When type casting is required in the API query, request or response, a fully
+qualified type name is represented as concatenation of a namespace and a type
+name. For consistent user experience namespaces should be aligned with the corresponding API category path segment.
+
+
+## When to Use this Pattern
+
+API resource grouping creates a user-friendly experience keeping all resources
+for a specific feature close together. It also allows generating smallersmaller downloadable client components 
+and limits the length of IDE prompts such as Intellisense.
+
+We recommend creating a new namespace when a new top-level API category is
+introduced.
+
+## Issues and Considerations
+
+1.  Microsoft Graph consistency requirements discourage using the same type
+    names for different concepts even within different namespaces. Microsoft
+    Graph type names must be descriptive and unique within the API
+    surface without requiring  full qualification.
+
+2.  Namespace must be consistent with API category in the navigation path according to [Microsoft Graph REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/graph/GuidelinesGraph.md#uniform-resource-locators-urls)
```

Comment:
Should me have this a relative URL so the link won't break in a different branch?
```suggestion
2.  Namespace must be consistent with API category in the navigation path according to [Microsoft Graph REST API Guidelines](GuidelinesGraph.md#uniform-resource-locators-urls)
```